src/utils/tests/test_compose_executor.py

```
@@ -0,0 +1,1087 @@
+        spec_text = textwrap.dedent('''\
+            workflow:
+              name: ok
+              tasks:
+              - name: task
+                image: ubuntu:24.04
+                command: ["echo"]
+                credentials:
+                  my-secret: NGC_API_KEY
+        ''')
+        executor = ComposeExecutor(
+            work_dir='/tmp/unused', credentials={'my-secret': '/tmp/secret-dir'})
```

Ruff (0.15.15)

[error] Probable insecure usage of temporary file or directory: "/tmp/unused"

(S108)